cpp/src/strings/json/json_to_array.cu

```diff
@@ -0,0 +1,563 @@
+  {
+    json_string value;
+    parse_whitespace();
+    if (*pos != ':') {
+      printf("Error json format is't colon character.\n");
+    }
+    pos++;
+    parse_whitespace();
+    char c = *pos;
+    if (c == '\"') {
+      value.str = ++pos;
+      char const* quo_pos = find_char_pos('\"');
+      value.len = quo_pos - value.str;
+      pos = ++quo_pos;
+      parse_whitespace();
+      char const* comma_pos = find_char_pos(',');
+      if (*pos == '}' && !eof(pos))
+      {
+        finish_obj = true;
+        finish_arr = false;
+      }
+      else if (eof(pos))
+      {
+        finish_obj = true;
+        finish_arr = true;
+      }
+      else{
+        finish_obj = false;
+        finish_arr = false;
+        pos = comma_pos;
+      }      
+      pos++;
+    }
+    else if (c == '{' || c == '[') {
+      value.str = pos;
+      int obj_count = 0;
+      int arr_count = 0;
+      while (!eof(pos)) {
+        char c = *pos++;
+        switch (c) {
+          case '{': obj_count++; break;
+          case '}': obj_count--; break;
+          case '[': arr_count++; break;
+          case ']': arr_count--; break;
+          default: break;
+        }
+        if (obj_count == 0 && arr_count == 0) break;
+      }
+      value.len = pos - value.str;
+      parse_whitespace();
+      char const* comma_pos = find_char_pos(',');
+      pos = comma_pos;
+      if (eof(pos))
+        finish_obj = true;
+      pos++;
+    }
+    else if (is_digit(c)) {
+      value.str = pos;
+      char const* comma_pos = find_char_pos(',');
+      char const* last_value_pos = find_last_validity_pos();
+      value.len = last_value_pos - pos;
+      pos = ++last_value_pos;
+      parse_whitespace();
+      if (*pos == ',' && comma_pos == pos) { //obj inner
+        finish_obj = false;
+        finish_arr = false;
+        pos++;
+      }
+      else if (*pos == '}' && eof(comma_pos)) { //eof end
+        finish_obj = true;
+        finish_arr = true;
+      }
+      else if (*pos == '}'){ // obj end
+        pos = comma_pos;
+        pos++;
+        finish_obj = true;
+        finish_arr = false;
+      }
+    }
+    else {
+      printf("Error when extract value from json foramt.\n");
+    }
+    return value;
+  }
+  CUDA_HOST_DEVICE_CALLABLE char parse_char() { return *pos++; }
+
+  first_operator_type element;
+
+  private:
+  CUDA_HOST_DEVICE_CALLABLE bool parse_string(json_string& str, bool can_be_empty)
+  {
+    str.str = nullptr;
+    str.len = 0;
+
+    if (parse_whitespace()) {
+      if (*pos == '\"') {
+        const char* start = ++pos;
+        while (!eof()) {
+          if (*pos == '\"') {
+            str.str = start;
+            str.len = pos - start;
+            pos++;
+            return true;
+          }
+          pos++;
+        }
+      }
+    }
+    return can_be_empty ? true : false;
+  }
+
+  const char* cur_el_start;
+};
+
+namespace strings {
+namespace detail {
+
+namespace {
+using namespace cudf;
```

Comment:
Do not import namespaces, instead use namespace prefix (`cudf::`) whenever it is required.